virttest/utils_test/__init__.py

```diff
@@ -975,6 +975,11 @@ def env_check(self):
             logging.error("Unable to find python.")
             return False
         self.pip_bin = find_bin(self.session, ['pip', 'pip2', 'pip3'])
+        if self.pip_bin:
```

Comment:
Yes, this makes sense to me, but simple fact that pip is present doesn't mean it's required for the setup. How about moving this to line 1018? Also I'd suggest modifying the commit message to say something like `Add a step to update pip as some older pip versions report exit code 0 for failed installations` (the current version implies that without update it'd never show errors but people might be running it on systems with already fixed pip)

Reply:
Actually I am wrong as pip is used later in `env_check`, so this is the right place.